x-pack/plugins/security_solution/cypress/e2e/detection_rules/bulk_edit_rules_actions.cy.ts

```diff
@@ -0,0 +1,186 @@
+  createThresholdRule,
+  createNewTermsRule,
+  createSavedQueryRule,
+  createCustomRuleEnabled,
+} from '../../tasks/api_calls/rules';
+import { createSlackConnector } from '../../tasks/api_calls/connectors';
+
+import {
+  getEqlRule,
+  getNewThreatIndicatorRule,
+  getNewRule,
+  getNewThresholdRule,
+  getMachineLearningRule,
+  getNewTermsRule,
+} from '../../objects/rule';
+
+const expectedNumberOfCustomRulesToBeEdited = 7;
+// 7 custom rules of different types + 3 prebuilt.
+// number of selected rules doesn't matter, we only want to make sure they will be edited an no modal window displayed as for other actions
+const expectedNumberOfRulesToBeEdited = expectedNumberOfCustomRulesToBeEdited + 3;
+
+const expectedSlackMessage = 'Slack action test message';
+
+describe('Detection rules, bulk edit of rule actions', () => {
+  before(() => {
+    cleanKibana();
+    login();
+  });
+
+  beforeEach(() => {
+    deleteAlertsAndRules();
+    deleteConnectors();
+    esArchiverResetKibana();
+
+    createCustomRuleEnabled(getNewRule(), '1');
+    createEventCorrelationRule(getEqlRule(), '2');
+    createMachineLearningRule(getMachineLearningRule(), '3');
+    createCustomIndicatorRule(getNewThreatIndicatorRule(), '4');
+    createThresholdRule(getNewThresholdRule(), '5');
+    createNewTermsRule(getNewTermsRule(), '6');
+    createSavedQueryRule({ ...getNewRule(), savedId: 'mocked' }, '7');
+
+    createSlackConnector();
+  });
+
+  context('Restricted action privileges', () => {
+    it('User has no actions privileges', () => {
+      login(ROLES.hunter_no_actions);
+      visitWithoutDateRange(SECURITY_DETECTIONS_RULES_URL, ROLES.hunter_no_actions);
+      waitForRulesTableToBeLoaded();
+
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+
+      openBulkActionsMenu();
+
+      cy.get(ADD_RULE_ACTIONS_MENU_ITEM).should('be.disabled');
+    });
+  });
+
+  context('All actions privileges', () => {
+    beforeEach(() => {
+      visitWithoutDateRange(SECURITY_DETECTIONS_RULES_URL);
+      waitForRulesTableToBeLoaded();
+    });
+
+    it('Add rule actions to rules', () => {
+      const expectedActionFrequency = 'Daily';
+
+      loadPrebuiltDetectionRulesFromHeaderBtn();
+
+      // select both custom and prebuilt rules
+      selectNumberOfRules(expectedNumberOfRulesToBeEdited);
+      openBulkEditRuleActionsForm();
+
+      // ensure rule actions info callout displayed on the form
+      cy.get(RULES_BULK_EDIT_ACTIONS_INFO).should('be.visible');
+
+      pickActionFrequency(expectedActionFrequency);
+      addSlackRuleAction(expectedSlackMessage);
+
+      submitBulkEditForm();
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfRulesToBeEdited });
+
+      // check if rule has been updated
+      goToEditRuleActionsSettings();
+
+      assertSelectedActionFrequency(expectedActionFrequency);
+      assertSlackRuleAction(expectedSlackMessage);
+    });
+
+    it('Overwrite rule actions in rules', () => {
```

Comment:
As we're testing overwrite functionality, shouldn't we have rules with existing actions as input here? So we could check that their actions were actually overwritten.

Reply:
added